litespi/__init__.py

```diff
@@ -71,10 +70,57 @@ def __init__(self, phy, sys_clk_freq, with_mmap=True, with_master=True, mmap_end
                 master.source.connect(port_master.sink),
             ]
 
-        self.comb += [
-            crossbar.master.source.connect(phy.sink),
-            phy.source.connect(crossbar.master.sink),
-            phy.cs_n.eq(crossbar.cs_n),
-            phy.clkgen.div.eq(self.clk_divisor.storage),
-            phy.dummy_bits.eq(dummy_bits.storage)
-        ]
+        if clock_domain is not "sys":
+            self.comb += [
+                crossbar.tx_cdc.source.connect(phy.sink),
+                phy.source.connect(crossbar.rx_cdc.sink),
+            ]
+        else:
+            self.comb += [
+                crossbar.master.source.connect(phy.sink),
+                phy.source.connect(crossbar.master.sink),
+            ]
+
+
+class LiteSPI(Module, AutoCSR, AutoDoc, ModuleDoc):
+    """SPI Core and PHY instantiator.
+
+    The ``LiteSPI`` class provides a wrapper that can instantiate ``LiteSPICore``, adds optional clock domain and gather CSRs from both core modules and PHY.
+
+    Parameters
+    ----------
+    phy : Module
+        Module or object that contains PHY stream interfaces and a cs_n signal to connect the ``LiteSPICore`` to.
+
+    clk_freq : int
+        Frequency of a clock connected to LiteSPI.
+
+    clock_domain : str
+        Name of LiteSPI clock domain.
+
+    with_mmap : bool
+        Enables memory-mapped SPI flash controller.
+
+    with_master : bool
+        Enables register-operated SPI master controller.
+
+    mmap_endianness : string
+        If endianness is set to ``small`` then byte order of each 32-bit word comming MMAP core will be reversed.
+
+    Attributes
+    ----------
+    bus : Interface(), out
+        Wishbone interface for memory-mapped flash access.
+    """
+
+    def __init__(self, phy, clk_freq, clock_domain="sys", with_mmap=True, with_master=True, mmap_endianness="big"):
+        assert with_mmap or with_master
+
+        self.submodules.core = LiteSPICore(phy, clk_freq, clock_domain, with_mmap, with_master, mmap_endianness)
+        self.bus = self.core.bus
+
+        self._csrs = []
+        self._csrs = self.core.get_csrs() + phy.get_csrs()
```

Comment:
I think this `LiteSPI` wrapper is no longer needed since the PHY now can handle clock domains on its own

Reply:
I reverted to original state. So we have just `LiteSPI`. CSRs for PHY and cores are separated [link](https://github.com/litex-hub/litespi/pull/38/files#diff-de492c7b09d278e7cf3335d6b1f192c0R89-R94)